SOURCES/src/kernel/MAKETERM.MAK

```diff
@@ -169,16 +171,18 @@ swterm.sys:	mosddtsh.asm  page.inc
         exe2bin $* $*.sys
         del $*.exe
 
-srterm.sys :	srterm.asm  srterm.def  page.inc  options.inc  mosregs.inc \
+srterm.sys :	srterm.asm  ..\mos5src\srterm.def  page.inc  options.inc  mosregs.inc \

```

Comment:
Not sure using stuff from mos5src in kernel directory is the way to go. I think that we have to figure what should be built where as there are so many duplicate files between directories. See
~~~
ajb@polly:/clients/common/pcmos386.git/SOURCES/src$ find . -iname srterm.asm -ls
 10619869     84 -rw-r--r--   1 ajb      sphere      82718 Jan 16 17:38 ./mossrc1/SRTERM.ASM
 10619746     80 -rw-r--r--   1 ajb      sphere      80594 Jan 16 17:38 ./mos5src/SRTERM.ASM
 10619372     80 -rw-r--r--   1 ajb      sphere      80594 Jan 16 17:38 ./kernel/SRTERM.ASM
 10619097     80 -rw-r--r--   1 ajb      sphere      80594 Jan 16 17:38 ./jim/SRTERM.ASM
~~~
Are you sure that maketerm is supposed to be built in kernel directory, the MAKEMOS.BAT has it building out of mos5src?

Reply:
I don't know about maketerm, but Roeland confirmed that 5.02 was work in progress, never completed. And that seems to be the purpose of MOS5SRC. Why we would want to mix 5.01 code with incomplete work? MAKEMOS.BAT may have had it wrong all along. Where did it come from? Who created it?

Reply:
> I don't know about maketerm, but Roeland confirmed that 5.02 was work in progress, never completed. And that seems to be the purpose of MOS5SRC.

I don't know how the case can be made that the kernel directory is complete release 5.01 when it seems there are missing files that can be found elsewhere. 

> MAKEMOS.BAT may have had it wrong all along. Where did it come from? Who created it?

Easy enough to look and for now it's the best indication that kernel is the place to build `kernel` and mos5src is the place to build `term` `utils` `import` and `export` anything else has to be established case by case.

~~~
 git log MAKEMOS.BAT

<snip>

Author: Roeland Jansen <roeland.jansen69@gmail.com>
Date:   Sun Sep 3 13:47:42 2017 +0200

    added source code tree, batchfile to build code
~~~

It was there from the start.

Reply:
I would like to hear from Roeland why it was done that way. You can't be sure there won't be a mismatch between kernel vs. shell when the files are duplicated but not identical. Very dangerous to assume that's safe. If would be safer to copy any missing files back to kernel, and leave 5.02 alone.

Reply:
That's an assumption that kernel is 5.01 and mos5src is 5.02, look at this:
~~~
ajb@polly:/clients/common/pcmos386.git/SOURCES/src$ git grep 5\\\.02
kernel/MONITOR.ASM:msgmain      db      1,25,'PC-MOS System Monitor v5.02$'  ;@@xlat
kernel/MONITOR.ASM:msginstall db        13,10,'PC-MOS System Monitor v5.02 Installed.        ',13,10,'$'  ;@@xlat
kernel/MONITOR.ASM:msgalready db        13,10,'PC-MOS System Monitor v5.02 already Installed.       '  ;@@xlat
mos5src/MONITOR.ASM:msgmain     db      1,25,'PC-MOS System Monitor v5.02$'  ;@@xlat
mos5src/MONITOR.ASM:msginstall db       13,10,'PC-MOS System Monitor v5.02 Installed.        ',13,10,'$'  ;@@xlat
mos5src/MONITOR.ASM:msgalready db       13,10,'PC-MOS System Monitor v5.02 already Installed.       '  ;@@xlat
mos5src/MOS.ASM:SAH 08/31/92    Update to 5.02 version for HOLD COMx fix
mos5src/MOS.ASM:        db      'PC-MOS MOS Utility v5.02 (920831)     ',13,10           ;@@XLAT
mos5src/MOSADM.ASM:SAH 08/31/92 Update to 5.02 version for HOLD COMx fix
mos5src/MOSADM.ASM:     db      'PC-MOS MOS Administrator Utility v5.02 (920831)          ',13,10  ;@@xlat
mos5src/MOSDDTSH.ASM:   title   pcterm.sys (v5.02) - mos pc shadow terminal driver module
mos5src/MOSDDTSH.ASM:   title   swterm.sys (v5.02) - mos sherwood terminal driver module
mos5src/MOSDDTSH.ASM:   title   elterm.sys (v5.02) - mos emulink terminal driver module
mos5src/MOSDDTSH.ASM:   title   atterm.sys (v5.02) - mos at&t 605 terminal driver module
mos5src/MOSDDTSH.ASM:   title   w7term.sys (v5.02) - mos WYSE 370 terminal driver module
mos5src/MOSDDTSH.ASM:   title   trterm.sys (v5.02) - mos TR-170 terminal driver module
mos5src/MOSDDTSH.ASM:   title   v4term.sys (v5.02) - mos DEC VT420 terminal driver module
mos5src/MOSDDTSH.ASM:ddtmsg     db      'HGTERM terminal device driver installed V5.02 (920601)'  ;@@xlat
mos5src/MOSDDTSH.ASM:ddtmsg     db      'ELTERM terminal device driver installed V5.02 (920601)'  ;@@xlat
mos5src/MOSDDTSH.ASM:ddtmsg     db      'ATTERM terminal device driver installed V5.02 (920601)'  ;@@xlat
mos5src/MOSDDTSH.ASM:ddtmsg     db      'PCTERM terminal device driver installed V5.02 (920601)'  ;@@xlat
mos5src/MOSDDTSH.ASM:ddtmsg     db      'SHERWOOD terminal device driver installed V5.02 (920601)'  ;@@xlat
mos5src/MOSDDTSH.ASM:ddtmsg     db      'WYSE 370 terminal device driver installed V5.02 (920601)'  ;@@xlat
mos5src/MOSDDTSH.ASM:ddtmsg     db      'TR 170 terminal device driver installed V5.02 (920601)'  ;@@xlat
mos5src/MOSDDTSH.ASM:ddtmsg     db      'DEC VT420 terminal device driver installed V5.02 (920601)'  ;@@xlat
mos5src/MOXCPCOR.ASM:SAH 08/04/92    Version 5.02 Well appears that Great Plains is playing 
mos5src/_EMS.ASM:message  db    'PC-MOS LIM 4.0 Expanded Memory Driver v5.02 (920814) $          ' ;@@xlat
~~~

At least some 5.02 is in `kernel` directory, so why not `jim` directory it has 5.01 monitor.asm?

Reply:
Sounds like someone should do a code audit.  I'll go back to looking for the 2012 bug and let someone else tackle that.

Reply:
Someone should. But that's a big job, maybe too big.

Reply:
And who's Jim?